transport/src/test/java/io/netty/channel/embedded/CustomChannelId.java

```diff
@@ -17,21 +17,20 @@
 
 import io.netty.channel.ChannelId;
 
-public class CustomChannelId implements ChannelId {
+public class CustomChannelId implements ChannelId<CustomChannelId> {
+
     private static final long serialVersionUID = 1L;
 
     private int id;
 
-    public CustomChannelId(int id) {
+    CustomChannelId(int id) {
         this.id = id;
     }
 
     @Override
-    public int compareTo(ChannelId o) {
-        if (o instanceof CustomChannelId) {
-            return id - ((CustomChannelId) o).id;
-        }
-        return -1;
+    public int compareTo(CustomChannelId o) {
+        // do not subtract for comparison, it could overflow
+        return id < o.id ? -1 : (id == o.id ? 0 : 1);
```

Comment:
consider adding a `compare` to MathUtil for `int`. That way when we upgrade our java support it will be easier to identify and replace with the JDK methods.